skiko/src/awtMain/objectiveC/macos/DisplayLinkThrottler.mm

```diff
@@ -101,33 +88,28 @@ - (void)setupDisplayLinkForWindow:(NSWindow *)window {
 
     if (result != kCVReturnSuccess) {
         CVDisplayLinkRelease(displayLink);
-
-        [self handleDisplayLinkSetupFailure];
+        _displayLink = nil
         return;
     }
 
     result = CVDisplayLinkSetOutputCallback(displayLink, &displayLinkCallback, (__bridge void *)(self));
 
     if (result != kCVReturnSuccess) {
         CVDisplayLinkRelease(displayLink);
-
-        [self handleDisplayLinkSetupFailure];
+        _displayLink = nil
         return;
     }
 
     result = CVDisplayLinkStart(displayLink);
 
     if (result != kCVReturnSuccess) {
         CVDisplayLinkRelease(displayLink);
-
-        [self handleDisplayLinkSetupFailure];
+        _displayLink = nil
```

Comment:
Removed.

Also, we forget about `_displayLinkScreen` in a case of failure. Extracted creation of the display link to another function. Now, if the display link failed, we still remember the screen, and don't try to create it next time.

Question - is it correct, or we should try to create display link every frame?

Reply:
Let's deduce it from documentation:

[CVDisplayLinkStart](https://developer.apple.com/documentation/corevideo/1457193-cvdisplaylinkstart?language=objc) only fails if it's already running (impossible branch, in we documentation is correct).
[CVDisplayLinkSetOutputCallback](https://developer.apple.com/documentation/corevideo/1457096-cvdisplaylinksetoutputcallback?language=objc) there is no clear documentation on why it can fail.
[CVDisplayLinkCreateWithCGDisplay](https://developer.apple.com/documentation/corevideo/1456981-cvdisplaylinkcreatewithcgdisplay?language=objc) also doesn't contain clear reason for possible failure.

So both solutions (stop trying recreating CVDisplayLink for a given screen vs try every frame until it succeeds) are equally adequate shots in the dark with second one being more optimistic.

Reply:
Let's don't make assumptions then, and settle on something.

Is the PR good to merge?

Reply:
Yes, looks good, let's stick with the latest solution